go/vt/srvtopo/resilient_server.go

```diff
@@ -308,7 +310,8 @@ func (server *ResilientServer) GetSrvKeyspaceNames(ctx context.Context, cell str
 				server.counts.Add(errorCategory, 1)
 				if entry.insertionTime.IsZero() {
 					log.Errorf("GetSrvKeyspaceNames(%v, %v) failed: %v (no cached value, caching and returning error)", ctx, cell, err)
-
+				} else if strings.Contains(err.Error(), "deadline exceeded") {
```

Comment:
It will be better to check
```
ctx.Err() == context.DeadlineExceeded
```
instead of checking for the error string.

Reply:
gotcha